extensions/data-workspace/src/dataworkspace.d.ts

```diff
@@ -115,4 +120,29 @@ declare module 'dataworkspace' {
 		 */
 		element: any;
 	}
+
+	export interface IProjectAction {
+		/**
+		 * id and display name of the project action
+		 */
+		readonly id: string;
+
+		/**
+		 * icon path of the project action
+		 */
+		readonly icon?: string | vscode.Uri | { light: string | vscode.Uri, dark: string | vscode.Uri };
+
+		/**
+		 * Run context for each project action
+		 * @param treeItem The treeItem in a project's hierarchy, to be used to obtain a Project
+		 */
+		run(treeItem: WorkspaceTreeItem): Promise<any>;
```

Comment:
Might be useful to have this be a generic `IProjectAction<T>` and then have the return type here be `Promse<T>`

Reply:
Removed the return type altogether.